### ISSUE TITLE
Log states no host or ip set 

### ISSUE THREAD

**Issue by aenglish86** [hjdhjd/homebridge-unifi-protect2#40]
Hello. I have followed the instructions on the plugin page of hoobs, set the username and password as well as IP address. I’m using the udm pro for unifi protect and have tried both ip address’ with and without the port value of 7443. I’ve also tried with and without http and https. I’ve also tried pointing it to unifi.ui.com
![9AA13241-0582-4937-91A0-7E4279F52A0E](https://user-images.githubusercontent.com/37191963/90462778-3f61bf00-e0be-11ea-8910-347e6752899c.jpeg)

**Comment by aenglish86** [comment]
I can send you screenshots via email of the actual configuration page

**Comment by hjdhjd** [comment]
Sure. The error is that you haven't configured a controller. If you can post the relevant portion of your config.json as well, it would be helpful.

**Comment by aenglish86** [comment]
![1C2E02DC-3C52-4BF0-A4CA-717F1DFD6F00](https://user-images.githubusercontent.com/37191963/90463618-412c8200-e0c0-11ea-8cea-2dcc62144057.png)

**Comment by hjdhjd** [comment]
A few questions:

* What version of the plugin are you on?
* What are you using for your UniFi Protect controller? UCKgen2+, UDM-Pro, UNVR, etc?
* Can you verify that 192.168.1.1 takes you to the UniFi Protect webUI?

**Comment by aenglish86** [comment]
Plugin version Unifi Protect2
3.3.0 published 8/17/2020

I am using the UDM Pro for my unifi protect. As with the udm pro accessing the unit is only accessible via: unifi.ui.com

**Comment by hjdhjd** [comment]
Well, the good news is that it looks like your config.json is right.
The bad news is that you can't access UniFi Protect locally. That's necessary for this plugin to work.

If you go to https://192.168.1.1 that should take you to your UDM-Pro login page. When you login, if you look in the top right you should see 9 "dots"...clicking on that should allow you to go to your local UniFi Protect. You should be able to login there.

**Comment by aenglish86** [comment]
Ok, so when I go to https://192.168.1.1 it does take me to the login page which it has never done before in the past. But when I put that address into the config, it still says no host or ip has been configured in the hoobs logs

**Comment by hjdhjd** [comment]
No. **Your prior config was correct**. You shouldn't have modified it.

You need to confirm that **you**, not the plugin, can successfully login to your UniFi Protect installation. Once you do, you can turn your focus to the plugin.

**Comment by aenglish86** [comment]
here is my config file in its entirety minus passwords:

"platforms": [
        {
            "platform": "TplinkSmarthome",
            "plugin_map": {
                "plugin_name": "homebridge-tplink-smarthome"
            }
        },
        {
            "platform": "Camera-ffmpeg",
            "plugin_map": {
                "plugin_name": "homebridge-camera-ffmpeg"
            },
            "name": "Camera FFmpeg"
        },
        {
            "platform": "UniFi Protect",
            "plugin_map": {
                "plugin_name": "homebridge-unifi-protect2"
            },
            "debug": false,
            "controllers": [
                {
                    "platform": "UniFi Protect",
                    "controllers": [
                        {
                            "address": "192.168.1.1",
                            "username": "aenglish86",
                            "password": "password"
                        }
                    ]
                }
            ]
        }
    ]
}

and here is the log:

2020-08-17, 7:54:40 PM Got SIGINT, shutting down Bridge...
2020-08-17, 7:54:45 PM Loaded plugin "homebridge-camera-ffmpeg".
2020-08-17, 7:54:45 PM Loaded plugin "homebridge-chamberlain".
2020-08-17, 7:54:46 PM Loaded plugin "homebridge-tplink-smarthome".
2020-08-17, 7:54:46 PM Loaded plugin "homebridge-unifi-protect2".
2020-08-17, 7:54:46 PM [TplinkSmarthome] homebridge-tplink-smarthome v5.1.0, node v12.16.3, homebridge v3.2.6
2020-08-17, 7:54:46 PM [UniFi Protect] No host or IP address has been configured.
2020-08-17, 7:54:46 PM [TplinkSmarthome] Configuring cached accessory: [Kids Bedroom] 800640B417ABB297FD65A2A8C91FDBEF1A75B6D4 01ef84df-6376-4f09-a926-f0cb8f5a2c9d
2020-08-17, 7:54:46 PM [TplinkSmarthome] Configuring cached accessory: [Laundry Room] 8006551926FEC0E86030E84D9B6DEF681A7587B4 ef11bec1-d034-4090-86c1-9a08f31e9c35
2020-08-17, 7:54:46 PM Bridge is running on port 51826.
2020-08-17, 7:54:46 PM [TplinkSmarthome] New Device Online: [Laundry Room] plug [8006551926FEC0E86030E84D9B6DEF681A7587B4] 192.168.1.61 9999
2020-08-17, 7:54:46 PM [TplinkSmarthome] Adding: [Laundry Room] plug [8006551926FEC0E86030E84D9B6DEF681A7587B4]
2020-08-17, 7:54:46 PM [TplinkSmarthome] New Device Online: [Kids Bedroom] plug [800640B417ABB297FD65A2A8C91FDBEF1A75B6D4] 192.168.1.203 9999
2020-08-17, 7:54:46 PM [TplinkSmarthome] Adding: [Kids Bedroom] plug [800640B417ABB297FD65A2A8C91FDBEF1A75B6D4]

If it couldn't connect to the controller wouldn't it throw some kind of error instead of saying no host or ip configured?

**Comment by hjdhjd** [comment]
Your configuration is in fact incorrect. Try this:

```
"platforms": [
        {
            "platform": "TplinkSmarthome",
            "plugin_map": {
                "plugin_name": "homebridge-tplink-smarthome"
            }
        },
        {
            "platform": "Camera-ffmpeg",
            "plugin_map": {
                "plugin_name": "homebridge-camera-ffmpeg"
            },
            "name": "Camera FFmpeg"
        },
        {
            "platform": "UniFi Protect",
            "plugin_map": {
                "plugin_name": "homebridge-unifi-protect2"
            },
            "debug": false,
            "controllers": [
              {
                  "address": "192.168.1.1",
                  "username": "aenglish86",
                  "password": "password"
               }
           ]
        }
    ]
}
```

**Comment by aenglish86** [comment]
copied and pasted the updated config file you posted for me to try and i still get this:

2020-08-17, 8:17:20 PM HOOBS listening on port 80.
2020-08-17, 8:17:23 PM Loaded plugin "homebridge-camera-ffmpeg".
2020-08-17, 8:17:23 PM Loaded plugin "homebridge-chamberlain".
2020-08-17, 8:17:23 PM Loaded plugin "homebridge-tplink-smarthome".
2020-08-17, 8:17:24 PM Loaded plugin "homebridge-unifi-protect2".
2020-08-17, 8:17:24 PM [TplinkSmarthome] homebridge-tplink-smarthome v5.1.0, node v12.16.3, homebridge v3.2.6
2020-08-17, 8:17:24 PM [UniFi Protect] No host or IP address has been configured.

**Comment by hjdhjd** [comment]
@aenglish86 I'm afraid I'm at the end of my ability to help you troubleshoot this on HOOBS. You might want to look at a regular homebridge install instead. Perhaps someone who is more familiar with HOOBS can provide some additional insight.

**Comment by zellunit** [comment]
i have exactly the same problem. has worked for 4 months flawlessly with 8 cameras then as soon as the last version bumped this happened @hjdhjd

**Comment by Altivec-Dan** [comment]
Have you tried replacing your username with email address?  Unifi Protect login has been flaky for me if not using the full email address.

**Comment by Archemedees** [comment]
I realize this is an old thread but if anyone else on hoobs has a similar experience. If you follow the instructions for install for homebridge you'll end up with this error. Hoobs already puts most of the config together for you. the only thing you need to add to the config through the config editor for the Unifi Protect Plugin is this: 

![image](https://user-images.githubusercontent.com/10063756/114642351-c55ff500-9ca1-11eb-9998-aad4d3519204.png)

You'll notice the platform callouts, the controller callout etc are all missing, Hoobs adds those in the config afterwards and if you put them here you'll be entering them twice and thus no IP is found. 

Not sure if this is the same issue but it solved it for me.
